src/vs/workbench/contrib/chat/browser/chatEditing/simpleBrowserEditorOverlay.ts

```diff
@@ -0,0 +1,299 @@
+
+	getDomNode(): HTMLElement {
+		return this._domNode;
+	}
+
+	hide() {
+		this._showStore.clear();
+	}
+}
+
+class SimpleBrowserOverlayController {
+
+	private readonly _store = new DisposableStore();
+
+	private readonly _domNode = document.createElement('div');
+
+	constructor(
+		container: HTMLElement,
+		group: IEditorGroup,
+		@IInstantiationService instaService: IInstantiationService,
+		@IChatService chatService: IChatService,
+		@IChatEditingService chatEditingService: IChatEditingService,
+		@IInlineChatSessionService inlineChatService: IInlineChatSessionService
+	) {
+
+		this._domNode.classList.add('chat-editing-editor-overlay');
+		this._domNode.style.position = 'absolute';
+		this._domNode.style.bottom = `5px`;
+		this._domNode.style.right = `5px`;
+		this._domNode.style.zIndex = `100`;
+
+		const widget = instaService.createInstance(SimpleBrowserOverlayWidget, group, container);
+		this._domNode.appendChild(widget.getDomNode());
+		this._store.add(toDisposable(() => this._domNode.remove()));
+		this._store.add(widget);
+
+		const show = () => {
+			if (!container.contains(this._domNode)) {
+				container.appendChild(this._domNode);
+			}
+		};
+
+		const hide = () => {
+			if (container.contains(this._domNode)) {
+				widget.hide();
+				this._domNode.remove();
+			}
+		};
+
+		const activeEditorSignal = observableSignalFromEvent(this, Event.any(group.onDidActiveEditorChange, group.onDidModelChange));
+
+		const activeUriObs = derivedOpts({ equalsFn: isEqual }, r => {
+
+			activeEditorSignal.read(r); // signal
+
+			const editor = group.activeEditorPane;
+			if (editor?.input.editorId === 'mainThreadWebview-simpleBrowser.view') {
+				const uri = EditorResourceAccessor.getOriginalUri(editor?.input, { supportSideBySide: SideBySideEditor.PRIMARY });
+				return uri;
+			}
+			return undefined;
+		});
+
+		this._store.add(autorun(r => {
+
+			const data = activeUriObs.read(r);
+
+			if (!data) {
+				hide();
+				return;
+			}
+
+			// widget.show();
+			show();
+
+
+		}));
+	}
+
+	dispose(): void {
+		this._store.dispose();
+	}
+}
+
+export class SimpleBrowserOverlay implements IWorkbenchContribution {
+
+	static readonly ID = 'chat.simpleBrowser.overlay';
+
+	private readonly _store = new DisposableStore();
+
+	constructor(
+		@IEditorGroupsService editorGroupsService: IEditorGroupsService,
+		@IInstantiationService instantiationService: IInstantiationService,
+	) {
+
+		const editorGroups = observableFromEvent(
+			this,
+			Event.any(editorGroupsService.onDidAddGroup, editorGroupsService.onDidRemoveGroup),
+			() => editorGroupsService.groups
+		);
+
+		const overlayWidgets = new DisposableMap<IEditorGroup>();
+
+		this._store.add(autorun(r => {
+
+			const toDelete = new Set(overlayWidgets.keys());
+			const groups = editorGroups.read(r);
+
+
+			for (const group of groups) {
+
+				if (!(group instanceof EditorGroupView)) {
+					// TODO@jrieken better with https://github.com/microsoft/vscode/tree/ben/layout-group-container
```

Comment:
🙈 dupes the todo I should have taken care of ;-). @bpasero maybe it's time we make this more "offical"